src/net_processing.cpp

```diff
@@ -321,10 +321,15 @@ struct CNodeState {
      */
     bool fSupportsDesiredCmpctVersion;
 
-    /** State used to enforce CHAIN_SYNC_TIMEOUT
-      * Only in effect for outbound, non-manual, full-relay connections, with
-      * m_protect == false
-      * Algorithm: if a peer's best known block has less work than our tip,
+    /** State used to enforce CHAIN_SYNC_TIMEOUT and EXTRA_PEER_CHECK_INTERVAL logics.
+      *
+      * Both are only in effect for outbound, non-manual, non-protected connections.
+      * Any peer protected (m_protect = true) is not chosen for eviction. Block-relay
+      * peers are always excluded from protection. A peer is marked as protected under
+      * the condition of receiving at least a valid connecting header, having a better
+      * chain than us and not having reached yet MAX_OUTBOUND_PEERS_TO_PROTECT_FROM_DISCONNECT.
```

Comment:
`and not having reached yet MAX_OUTBOUND_PEERS_TO_PROTECT_FROM_DISCONNECT` makes it sound like *the peer* is not having reached, not us. I suggest this:

A peer is marked as protected if
      - it gave us a valid connecting header
      - it has a better chain than we have
      - we haven't reached MAX_OUTBOUND_PEERS_TO_PROTECT_FROM_DISCONNECT yet